src/dune_engine/scheduler.ml

```diff
@@ -677,21 +694,45 @@ module Run_once : sig
 end = struct
   type run_error =
     | Got_signal
-    | Files_changed
     | Never
     | Exn of Exn_with_backtrace.t
 
   exception Abort of run_error
 
-  let iter () =
-    let count = Event.pending_jobs () in
-    if count = 0 then
+  (** This function is the heart of the scheduler. It makes progress in
+      executing fibers by doing the following:
+
+      - notifying completed jobs
+      - starting cancellations
+      - terminating the scheduler on signals *)
+  let rec iter (t : t) =
+    (* In polling mode, we always wait until the next job starts, so there's no
```

Comment:
This `Abort Never` is reported when a build deadlocks, right?
It's seems surprising to me to just skip deadlock detection in polling mode. Shouldn't this only be allowed when the build status is "Waiting_for_file_changes", or something like that?

Reply:
That seems fine, let me experiment with doing that.